docs/dev.rst

```diff
@@ -1,20 +1,513 @@
+but allows you to be more selective regarding which components to retrieve when compared to the ``dx_pull`` task. 
+Manual tracking of component versions also allows for the possibility of retrieving some changes into one directory,
+and then running the task again to retrieve other changes into a different directory.
+ 
+.. note::
+
+    CumulusCI has multiple tasks for retrieving Metadata from an org environment.
+    For a comprehensive list, see the list of :ref:`retrieve changes` tasks in the cheat sheet.
 
-Track and retrieve changes
---------------------------
+When you are ready to capture changes in an org, run the ``retrieve_changes`` task:
 
-Run Apex tests
+.. code-block:: console
+
+    $ cci task run retrieve_changes --org dev
+
+The task accepts ``include``, ``exclude``, and ``types`` options for filtering the
+list of changed components, for scenarios where you don't want to retrieve all changed components.
+
+After the metadata has been retrieved, the org snapshot will be updated so 
+that the retrieved components will no longer be included in ``list_changes``.
+You can avoid this by setting the ``snapshot`` option to False.
+
+By default, changes are retrieved into the ``src`` directory when using Metadata source format,
+or the default ``sfdx`` package directory (``force-app``) when using ``sfdx`` source format.
+You can retrieve into a different location using the ``path`` option:
+
+.. code-block:: console
+
+    $ cci task run retrieve_changes --org dev --path unpackaged/config/qa
+
+
+
+Push Changes
+------------
+Developers rarely edit code directly in an org environment, but instead use an editor or IDE like VSCode or IntelliJ.
+After code (or other Metadata) in an editor you will need to push these changes from your project's local repository to the target org.
+
+If your project uses the ``sfdx`` source format then you can use the ``dx_push`` task:
+
+.. code-block:: console
+
+    $ cci task run dx_push
+
+If you project uses the Metadata source format you can use the ``deploy`` task:
+
+.. code-block:: console
+
+    $ cci task run deploy 
+
+.. note::
+   
+   The ``deploy`` task has *many* options for handling a number of different scenarios.
+   For a complete list see the reference documentation for the `deploy task <TODO>`_.
+
+
+
+Run Apex Tests
 --------------
+CumulusCI allows you to easily execute Apex tests in an org:
+
+.. code-block:: console
+
+    $ cci task run run_tests --org <org_name>
+
+.. note::
 
-Set up a QA org
+    This task has many options for running tests in a variety of ways.
+    For more information on options and examples see the reference documentation for `run_tests <TODO>`_.
+
+
+
+Set Up a QA Org
 ---------------
+There is flow named ``qa_org`` that is specific to setting up
+org environments that allow quality engineers to test features quickly, and easily.
+The ``qa_org`` runs the specialized ``config_qa`` task after deploying the projects
+(unmanaged) Metadata to the org.
```

Comment:
Again, I'd suggest breaking it down:

The qa_org flow sets up a scratch org to use for testing and QA. It does this in three phases which should look familiar from the dev_org flow:

1. Install dependencies and deploy any unpackaged metadata from the unpackaged/pre folder, using the dependencies flow.
2. Deploy the project's main package as an unmanaged package, using the deploy_unmanaged flow.
3. Perform additional unpackaged configuration steps. Here is where ``qa_org`` can differ from ``dev_org``, because it uses the ``config_qa`` flow for this phase instead of ``config_dev``.